src/lib/add-on/clients/add-on_auto.rb

```diff
@@ -260,7 +267,7 @@ def create_source(add_on, product, media_url)
     #
     # @return [String] absolute media url or empty string
     def media_url_for(add_on)
-      media_url = add_on.fetch("media_url", "")
+      media_url = addon_url(add_on)
```

Comment:
`addon_url(add_on)` is a bad name IMHO, especially when confronted with `media_url_for(add_on)`

How about renaming them to `stripped_addon_url` and `absolute_addon_url`? (or with `media` instead of `addon`)